sklearn/tree/_criterion.pyx

```diff
@@ -1335,10 +1333,12 @@ cdef class Poisson(RegressionCriterion):
         """
         cdef const DOUBLE_t[:, ::1] y = self.y
         cdef const DOUBLE_t[:] sample_weight = self.sample_weight
+        cdef const SIZE_t[:] sample_indices = self.sample_indices
 
         cdef DOUBLE_t y_mean = 0.
         cdef DOUBLE_t poisson_loss = 0.
         cdef DOUBLE_t w = 1.0
+        cdef SIZE_t i, k, p
```

Comment:
FYI @jshinm you forgot to define this in your PR to address Julien's comments. In Cython, all variables must be typed, meaning the type is defined apriori to usage.

Reply:
Cython will always surprise me. I am not really sure how did it infer previously and create those variables for us.